packages/playground/src/components/node_details.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+  <td :colspan="columnsLen" v-if="dNodeLoading" style="text-align: center">
+    <div class="pa-1">
+      <v-progress-circular indeterminate model-value="20" :width="3"></v-progress-circular>
+    </div>
+  </td>
+  <td :colspan="columnsLen" v-else-if="dNodeError" style="text-align: center">
+    <div class="pt-4">
+      <v-alert
+        variant="tonal"
+        class="d-flex justify-between"
+        color="#f44336"
+        dense
+        outlined
+        type="error"
+        style="text-align: center"
+      >
+        <div style="display: flex; align-items: center">Failed to retrieve Node details.</div>
+        <template v-slot:append>
+          <v-icon @click="getNodeDetails" style="cursor: pointer">mdi-reload</v-icon>
+        </template>
+      </v-alert>
+    </div>
+  </td>
+
+  <td :colspan="columnsLen" v-else>
+    <v-row class="ma-2">
+      <v-col :cols="getColSize">
+        <div class="mt-3">
+          <card-details
+            :loading="false"
+            title="Node Resources"
+            icon="mdi-harddisk"
+            :items="NodeResourceFields"
+          ></card-details>
+        </div>
+      </v-col>
+
+      <v-col :cols="getColSize">
+        <div class="mt-3">
+          <card-details
+            :loading="false"
+            title="Location"
+            icon="mdi-map-marker"
+            :items="CountryResourceFields"
+          ></card-details>
+        </div>
+      </v-col>
+
+      <v-col :cols="getColSize" :class="{ 'mt-n8': getColSize() === 6 }">
+        <div class="mt-3">
+          <card-details
+            :loading="false"
+            title="Farm details"
+            icon="mdi-silo"
+            :items="FarmResourceFields"
+          ></card-details>
+        </div>
+      </v-col>
+    </v-row>
+    <v-row>
+      <v-col v-if="item.value.num_gpu > 0" cols="getColSize" style="min-width: 600px; min-height: 400px">
+        <div class="mt-3">
+          <GPUDetailsCard :node="node" />
+        </div> </v-col
+    ></v-row>
+  </td>
+</template>
+
+<script setup lang="ts">
+import type { GridNode } from "@threefold/gridproxy_client";
+import { onMounted, ref, watch } from "vue";
+
+import { getNode } from "@/explorer/utils/helpers";
+import type { NodeDetailsCard } from "@/explorer/utils/types";
+
+import { gridProxyClient } from "../clients";
+import { nodeInitializer } from "../explorer/utils/types";
+import toTeraOrGigaOrPeta from "../utils/toTeraOrGegaOrPeta";
+
+const dNodeError = ref(false);
+const dNodeLoading = ref(true);
+const farmName = ref("");
+const publicIps = ref(0);
+const NodeResourceFields = ref<NodeDetailsCard[]>();
+const CountryResourceFields = ref<NodeDetailsCard[]>();
+const FarmResourceFields = ref<NodeDetailsCard[]>();
+const node = ref<GridNode>(nodeInitializer);
+
+const props = defineProps({
+  item: {
+    required: true,
+    type: null,
```

Comment:
What is this item of type null?

Reply:
The item contains data of node at specific row.

Reply:
So `node` with node type will be better.